tests/test_import.py

```diff
@@ -0,0 +1,34 @@
+import importlib
+import warnings
+import pytest
+
+def test_import_kedro_with_no_official_support_raise_error(mocker, monkeypatch):
+    """Test importing kedro should fails with python>=3.11 should fails"""
+    import kedro
+    with mocker.patch("kedro.sys"):
```

Comment:
As they stand, the result of these tests depends on the Python version they're ran on, so we are forced to patch `kedro.sys.version_info` I'd say (both with the `(3, 11)` case and the `(3, 10)` one)

Reply:
I understand this, two questions I have here:
1. I expect `mocker.patch` is same as `unittest.mock.patch`, but `mocker.patch("kedro.sys.version_info")` is not allowed but `patch("kedro.sys.version_info")` is allowed. I guess I won't be too obsess with `pytest` here and just use `unittest.mock`
2. All tests are failing which I am not sure why.

I will make parmaterize it with success(3.10) and fail case (3.11) later, but need to make 2. works first.

Reply:
TIL: https://pytest-mock.readthedocs.io/en/latest/remarks.html#usage-as-context-manager

Reply:
Ah yes, but I keep switching between mock and mocker so I mess this up.

I don't think the context manager is the problem here since removing the with have the same result, i could explain more next Monday.

Reply:
I pushed a proposal, let me know what you think.

Indeed there were some weird things with reloading the module, and the interaction between the various warning control options. For example, `pytest.mark.filterwarnings` was not setting `sys.warnoptions`, so it was not working as intended and I had to mock instead. Setting the environment variable with monkeypatch didn't work for me either. And finally, catching the `KedroPythonVersionWarning` didn't work, so I used the parent class instead.